nav2_costmap_2d/src/costmap_2d_ros.cpp

```diff
@@ -71,6 +71,15 @@ Costmap2DROS::Costmap2DROS()
     "nav2_costmap_2d::InflationLayer"}
 {
   declare_parameter("map_topic", rclcpp::ParameterValue(std::string("map")));
+  /**
+   * @brief
+   * If launched as an independent node, set `is_lifecycle_follower_  = true`.
+   * Thus, it would react to rcl_preshutdown.
+   * Else if launched in a thread created by another LifecycleNode(its parent), set
+   * `is_lifecycle_follower_  = false`. It would not react to rcl_preshutdown anymore
+   * and its lifecycle state is controlled by its parent.
+   */
+  get_parameter("is_lifecycle_follower", is_lifecycle_follower_);
```

Comment:
Not a parameter. Just set it to `false`, we know its false if this constructor is used. Its used when launched as an individual node and/or composition node. The other constructor's given their arguments require it to be a module of another system.